// @vue-flow/controls/dist/vue-flow-controls.mjs
import { openBlock as t, createElementBlock as i, renderSlot as n, createElementVNode as h, defineComponent as y, toRef as r, computed as b, createBlock as a, unref as l, withCtx as v, Fragment as M, createVNode as $, resolveDynamicComponent as d, createCommentVNode as u } from "vue";
import { useVueFlow as x, Panel as F } from "@vue-flow/core";
const O = {
  name: "ControlButton"
}, A = (o, e) => {
  const s = o.__vccOpts || o;
  for (const [_, w] of e)
    s[_] = w;
  return s;
}, P = { class: "vue-flow__controls-button" };
function Z(o, e, s, _, w, V) {
  return t(), i("button", P, [
    n(o.$slots, "default")
  ]);
}
const f = /* @__PURE__ */ A(O, [["render", Z]]), D = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 32 32"
}, N = /* @__PURE__ */ h("path", { d: "M32 18.133H18.133V32h-4.266V18.133H0v-4.266h13.867V0h4.266v13.867H32z" }, null, -1), S = [
  N
];
function E(o, e) {
  return t(), i("svg", D, S);
}
const L = { render: E }, R = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 32 5"
}, U = /* @__PURE__ */ h("path", { d: "M0 0h32v4.2H0z" }, null, -1), j = [
  U
];
function q(o, e) {
  return t(), i("svg", R, j);
}
const G = { render: q }, J = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 32 30"
}, K = /* @__PURE__ */ h("path", { d: "M3.692 4.63c0-.53.4-.938.939-.938h5.215V0H4.708C2.13 0 0 2.054 0 4.63v5.216h3.692V4.631zM27.354 0h-5.2v3.692h5.17c.53 0 .984.4.984.939v5.215H32V4.631A4.624 4.624 0 0 0 27.354 0zm.954 24.83c0 .532-.4.94-.939.94h-5.215v3.768h5.215c2.577 0 4.631-2.13 4.631-4.707v-5.139h-3.692v5.139zm-23.677.94a.919.919 0 0 1-.939-.94v-5.138H0v5.139c0 2.577 2.13 4.707 4.708 4.707h5.138V25.77H4.631z" }, null, -1), Q = [
  K
];
function T(o, e) {
  return t(), i("svg", J, Q);
}
const W = { render: T }, X = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 25 32"
}, Y = /* @__PURE__ */ h("path", { d: "M21.333 10.667H19.81V7.619C19.81 3.429 16.38 0 12.19 0 8 0 4.571 3.429 4.571 7.619v3.048H3.048A3.056 3.056 0 0 0 0 13.714v15.238A3.056 3.056 0 0 0 3.048 32h18.285a3.056 3.056 0 0 0 3.048-3.048V13.714a3.056 3.056 0 0 0-3.048-3.047zM12.19 24.533a3.056 3.056 0 0 1-3.047-3.047 3.056 3.056 0 0 1 3.047-3.048 3.056 3.056 0 0 1 3.048 3.048 3.056 3.056 0 0 1-3.048 3.047zm4.724-13.866H7.467V7.619c0-2.59 2.133-4.724 4.723-4.724 2.591 0 4.724 2.133 4.724 4.724v3.048z" }, null, -1), oo = [
  Y
];
function to(o, e) {
  return t(), i("svg", X, oo);
}
const eo = { render: to }, no = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 25 32"
}, so = /* @__PURE__ */ h("path", { d: "M21.333 10.667H19.81V7.619C19.81 3.429 16.38 0 12.19 0c-4.114 1.828-1.37 2.133.305 2.438 1.676.305 4.42 2.59 4.42 5.181v3.048H3.047A3.056 3.056 0 0 0 0 13.714v15.238A3.056 3.056 0 0 0 3.048 32h18.285a3.056 3.056 0 0 0 3.048-3.048V13.714a3.056 3.056 0 0 0-3.048-3.047zM12.19 24.533a3.056 3.056 0 0 1-3.047-3.047 3.056 3.056 0 0 1 3.047-3.048 3.056 3.056 0 0 1 3.048 3.048 3.056 3.056 0 0 1-3.048 3.047z" }, null, -1), co = [
  so
];
function lo(o, e) {
  return t(), i("svg", no, co);
}
const ro = { render: lo }, ao = {
  name: "Controls"
}, ho = /* @__PURE__ */ y({
  ...ao,
  props: {
    showZoom: { type: Boolean, default: !0 },
    showFitView: { type: Boolean, default: !0 },
    showInteractive: { type: Boolean, default: !0 },
    fitViewParams: null,
    position: { default: () => "bottom-left" }
  },
  emits: ["zoomIn", "zoomOut", "fitView", "interactionChange"],
  setup(o, { emit: e }) {
    const s = x(), _ = r(s, "nodesDraggable"), w = r(s, "nodesConnectable"), V = r(s, "elementsSelectable"), p = r(s, "setInteractive"), z = r(s, "zoomIn"), g = r(s, "zoomOut"), C = r(s, "fitView"), m = b(() => _.value && w.value && V.value), k = () => {
      z.value(), e("zoomIn");
    }, H = () => {
      g.value(), e("zoomOut");
    }, I = () => {
      C.value(o.fitViewParams), e("fitView");
    }, B = () => {
      p.value(!m.value), e("interactionChange", !m.value);
    };
    return (c, io) => (t(), a(l(F), {
      class: "vue-flow__controls",
      position: o.position
    }, {
      default: v(() => [
        n(c.$slots, "top"),
        o.showZoom ? (t(), i(M, { key: 0 }, [
          n(c.$slots, "control-zoom-in", {}, () => [
            $(f, {
              class: "vue-flow__controls-zoomin",
              onClick: k
            }, {
              default: v(() => [
                n(c.$slots, "icon-zoom-in", {}, () => [
                  (t(), a(d(l(L))))
                ])
              ]),
              _: 3
            })
          ]),
          n(c.$slots, "control-zoom-out", {}, () => [
            $(f, {
              class: "vue-flow__controls-zoomout",
              onClick: H
            }, {
              default: v(() => [
                n(c.$slots, "icon-zoom-out", {}, () => [
                  (t(), a(d(l(G))))
                ])
              ]),
              _: 3
            })
          ])
        ], 64)) : u("", !0),
        o.showFitView ? n(c.$slots, "control-fit-view", { key: 1 }, () => [
          $(f, {
            class: "vue-flow__controls-fitview",
            onClick: I
          }, {
            default: v(() => [
              n(c.$slots, "icon-fit-view", {}, () => [
                (t(), a(d(l(W))))
              ])
            ]),
            _: 3
          })
        ]) : u("", !0),
        o.showInteractive ? n(c.$slots, "control-interactive", { key: 2 }, () => [
          o.showInteractive ? (t(), a(f, {
            key: 0,
            class: "vue-flow__controls-interactive",
            onClick: B
          }, {
            default: v(() => [
              l(m) ? n(c.$slots, "icon-unlock", { key: 0 }, () => [
                (t(), a(d(l(ro))))
              ]) : u("", !0),
              l(m) ? u("", !0) : n(c.$slots, "icon-lock", { key: 1 }, () => [
                (t(), a(d(l(eo))))
              ])
            ]),
            _: 3
          })) : u("", !0)
        ]) : u("", !0),
        n(c.$slots, "default")
      ]),
      _: 3
    }, 8, ["position"]));
  }
});
export {
  f as ControlButton,
  ho as Controls
};
